sdk/communication/azure-communication-chat/azure/communication/chat/_shared/models.py

```diff
@@ -6,210 +6,151 @@
+
     def __init__(self, identifier):
+        # type: (str) -> None
         self.raw_id = identifier
+        self.properties = {}
 
-class CommunicationIdentifierModel(msrest.serialization.Model):
-    """Communication Identifier Model.
-    All required parameters must be populated in order to send to Azure.
-    :param kind: Required. Kind of Communication Identifier.
-    :type kind: CommunicationIdentifierKind
-    :param id: Full id of the identifier.
-    :type id: str
-    :param phone_number: phone number in case the identifier is a phone number.
-    :type phone_number: str
-    :param is_anonymous: True if the identifier is anonymous.
-    :type is_anonymous: bool
-    :param microsoft_teams_user_id: Microsoft Teams user id.
-    :type microsoft_teams_user_id: str
-    :param communication_cloud_environment: Cloud environment that the user belongs to.
-    :type communication_cloud_environment: CommunicationCloudEnvironment
-    """
 
-    _validation = {
-        'kind': {'required': True},
-    }
-
-    _attribute_map = {
-        'kind': {'key': 'kind', 'type': 'str'},
-        'id': {'key': 'id', 'type': 'str'},
-        'phone_number': {'key': 'phoneNumber', 'type': 'str'},
-        'is_anonymous': {'key': 'isAnonymous', 'type': 'bool'},
-        'microsoft_teams_user_id': {'key': 'microsoftTeamsUserId', 'type': 'str'},
-        'communication_cloud_environment': {'key': 'communicationCloudEnvironment', 'type': 'str'},
-    }
-
-    def __init__(
-        self,
-        **kwargs
-    ):
-        super(CommunicationIdentifierModel, self).__init__(**kwargs)
-        self.kind = kwargs['kind']
-        self.id = kwargs.get('id', None)
-        self.phone_number = kwargs.get('phone_number', None)
-        self.is_anonymous = kwargs.get('is_anonymous', None)
-        self.microsoft_teams_user_id = kwargs.get('microsoft_teams_user_id', None)
-        self.communication_cloud_environment = kwargs.get('communication_cloud_environment', None)
-
-class _CaseInsensitiveEnumMeta(EnumMeta):
-    def __getitem__(self, name):
-        return super().__getitem__(name.upper())
-
-    def __getattr__(cls, name):
-        """Return the enum member matching `name`
-        We use __getattr__ instead of descriptors or inserting into the enum
-        class' __dict__ in order to support `name` and `value` being both
-        properties for enum members (which live in the class' __dict__) and
-        enum members themselves.
-        """
-        try:
-            return cls._member_map_[name.upper()]
-        except KeyError:
-            raise AttributeError(name)
-
-class CommunicationIdentifierKind(with_metaclass(_CaseInsensitiveEnumMeta, str, Enum)):
-    """Communication Identifier Kind.
-    """
-    Unknown = "UNKNOWN"
-    CommunicationUser = "COMMUNICATIONUSER"
-    PhoneNumber = "PHONENUMBER"
-    CallingApplication = "CALLINGAPPLICATION"
-    MicrosoftTeamsUser = "MICROSOFTTEAMSUSER"
+MicrosoftTeamsUserProperties = TypedDict(
```

Comment:
Should this have `total=False`?

Reply:
I believe all the properties in the MicrosoftTeamsUserIdentifier are required - can you confirm @DominikMe?

Reply:
Both as inputs and outputs?

Reply:
They all have defined default values which get set inside the `MicrosoftTeamsUserIdentifier` constructor and the user doesn't instantiate the `MicrosoftTeamsUserProperties` dict themselves.

So they are optional on the identifier but then filled in the properties dictionary, this design seems reasonable to me.

If the service starts adding additional properties they will be ignored by the client on deserialization, which is good. If the service removes properties, we should throw because that's breaking. (see other thread about service response validation)